src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -466,12 +499,14 @@ WorkerCreateShard(Oid relationId, char *nodeName, uint32 nodePort,
 							 escapedDDLCommand);
 		}
 
-		queryResultList = ExecuteRemoteQuery(nodeName, nodePort, newShardOwner,
-											 applyDDLCommand);
-		if (queryResultList == NIL)
+		if (createInSeparateTransaction)
 		{
-			shardCreated = false;
-			break;
+			ExecuteRemoteQuery(nodeName, nodePort, newShardOwner,
+							   applyDDLCommand);
+		}
+		else
+		{
+			ExecuteCriticalRemoteCommand(connection, applyDDLCommand->data);
```

Comment:
I wonder whether we should keep the failover logic in `master_create_empty_shard`, since we have users that rely on that.